RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc

```diff
@@ -382,18 +504,19 @@ void TrackExtenderWithMTDT<TrackCollection>::produce(edm::Event& ev, const edm::
       pv = &(vtxH.product()->at(0));
```

Comment:
the vertex validity check should not be needed. Absence of a vertex in this context is more likely due to a misconfiguration of the module.
Perhaps a check for the collection size is still OK.